src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentProperty.java

```diff
@@ -44,12 +47,14 @@
 	private final boolean isScore;
 	private final boolean isParent;
 	private final boolean isId;
+	private final String annotatedFieldName;
```

Comment:
In the long run, we might leave caching of model-metadata away at `SimpleElasticsearchPersistentProperty` level and eventually introduce a  `CachingElasticsearchPersistentProperty` (see Mongo's [`CachingMongoPersistentProperty`](https://github.com/spring-projects/spring-data-mongodb/blob/3eb9d790b94e5a40660ef9c5cb262cbe684bbdcb/spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/mapping/CachingMongoPersistentProperty.java#L29)).

Reply:
thanks for the info; I'll have a look at how this is handled in Mongo if I find some time these days